examples/graphs/node2vec_movielens.py

```diff
@@ -0,0 +1,563 @@
+
+    example_weights = defaultdict(int)
+
+    # Iterate over all sequences (walks).
+    for sequence in tqdm(
+        sequences,
+        position=0,
+        leave=True,
+        desc=f"Generating postive and negative examples",
+    ):
+
+        # Generate positive and negative skip-gram pairs for a sequence (walk).
+        pairs, labels = keras.preprocessing.sequence.skipgrams(
+            sequence,
+            vocabulary_size=vocabulary_size,
+            window_size=window_size,
+            negative_samples=num_negative_smaples,
+        )
+
+        for idx in range(len(pairs)):
+            pair = pairs[idx]
+            label = labels[idx]
+            target, context = min(pair[0], pair[1]), max(pair[0], pair[1])
+            if target == context:
+                continue
+            entry = (target, context, label)
+            example_weights[entry] += 1
+
+    targets, contexts, labels, weights = [], [], [], []
+    for entry in example_weights:
+        weight = example_weights[entry]
+        target, context, label = entry
+
+        targets.append(target)
+        contexts.append(context)
+        labels.append(label)
+        weights.append(weight)
+
+    return np.array(targets), np.array(contexts), np.array(labels), np.array(weights)
+
+
+num_negative_smaples = 4
+targets, contexts, labels, weights = generate_examples(
+    sequences=walks,
+    window_size=num_steps,
+    num_negative_smaples=num_negative_smaples,
+    vocabulary_size=len(vocabulary),
+)
+
+"""
+Let's display the shapes of the outputs
+"""
+
+print(f"Targets shape: {targets.shape}")
+print(f"Contexts shape: {contexts.shape}")
+print(f"Labels shape: {labels.shape}")
+print(f"Weights shape: {weights.shape}")
+
+"""
+### Create tf.data.Dataset
+"""
+
+batch_size = 1024
+
+
+def create_dataset(targets, contexts, labels, weights, batch_size):
+    inputs = {
+        "target": targets,
+        "context": contexts,
+    }
+    dataset = tf.data.Dataset.from_tensor_slices((inputs, labels, weights))
+    dataset = dataset.shuffle(buffer_size=batch_size * 2)
+    dataset = dataset.batch(batch_size, drop_remainder=True)
+    return dataset
+
+
+dataset = create_dataset(
+    targets=targets,
+    contexts=contexts,
+    labels=labels,
+    weights=weights,
+    batch_size=batch_size,
+)
+
+"""
+## Train a Skip-gram Model
+
+Our skip-gram is a simple binary classification model that works as follows:
+1. An embedding is looked up for the `target` movie.
+2. An embedding is looked up for the `context` movie.
+3. The dot product is computed between these two embeddings.
+4. The result (after a sigmoid activation) is compared to the label.
+5. A binary crossentropy loss is used.
+"""
+
+learning_rate = 0.001
+embedding_dim = 50
+num_epochs = 10
+
+"""
+### Implement the model
+"""
+
+
+def create_model(vocabulary_size, embedding_dim):
+
+    inputs = {
+        "target": layers.Input(name="target", shape=(), dtype=tf.int32),
```

Comment:
Prefer using strings for dtypes, e.g. `"int32"`

Reply:
Done